### PR TITLE
Module 11

### PR DESCRIPTION
**Pull request by pcoulso1**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -30,20 +26,18 @@ script:
...
-- echo $DOCKERHUB_PWD | docker login --username $DOCKER_NAMESPACE --password-stdin
-- docker push $DOCKER_NAMESPACE/$DOCKER_APP:$TRAVIS_COMMIT
-- if [ "$TRAVIS_BRANCH" = "master" ]; then docker push $DOCKER_NAMESPACE/$DOCKER_APP; fi
+- docker run --rm -e FLASK_APP -e FLASK_ENV -e MONGO_URL -e MONGO_DEFAULT_DATABASE -e OAUTHLIB_INSECURE_TRANSPORT -e GITHUB_CLIENT_ID -e GITHUB_CLIENT_SECRET -e GITHUB_LOGON_REDIRECT $DOCKER_NAMESPACE/$DOCKER_APP-test tests
+- docker run --rm -e FLASK_APP -e FLASK_ENV -e MONGO_URL -e MONGO_DEFAULT_DATABASE -e OAUTHLIB_INSECURE_TRANSPORT -e GITHUB_CLIENT_ID -e GITHUB_CLIENT_SECRET -e GITHUB_LOGON_REDIRECT $DOCKER_NAMESPACE/$DOCKER_APP-test tests_e2e
+
+before_deploy:
+  - echo $DOCKERHUB_PWD | docker login --username $DOCKER_NAMESPACE --password-stdin
+  - docker push $DOCKER_NAMESPACE/$DOCKER_APP:$TRAVIS_COMMIT
+  - if [ "$TRAVIS_BRANCH" = "master" ]; then docker push $DOCKER_NAMESPACE/$DOCKER_APP; fi
```
Comment (JackMeadDev) [inline .travis.yml]: This condition should be handled by the `on` clause in the deploy step, unless you're expecting some branches to perform half of a deployment?

File: mongo_db/store.py [hunk 2]
```diff
@@ -19,7 +19,7 @@ def get_mongodb(self, db_name=Config().MONGO_DEFAULT_DATABASE):
 
             if self.MONGODB_CLIENT is None:
                 self.MONGODB_CLIENT = pymongo.MongoClient(
-                    f'mongodb+srv://{Config().MONGO_USER_NAME}:{Config().MONGO_PASSWORD}@{Config().MONGO_HOST}/{db_name}?retryWrites=true&w=majority')
+                    f'{Config().MONGO_URL}')
```
Comment (JackMeadDev) [inline mongo_db/store.py]: We could remove the string formatting here given that the URL is just being formatted as itself, but not important

**Review by JackMeadDev — APPROVED** [review]
This looks good, I can see the app is deployed to Azure and running - I saw errors after routing to Github for authorisation, but I can see that's working on your terraform infrastructure so I suspect just a mismatched environment variable and happy to leave that.
It would be nice to add some documentation to your readme mentioning that this infrastructure exists, and e.g. linking to the Azure URL

File: mongo_db/store.py (hunk 2 above)
Comment (pcoulso1) [reply mongo_db/store.py]: done

File: .travis.yml (hunk 1 above)
Comment (pcoulso1) [reply .travis.yml]: done

**Pull request merged**